Java/TelemetrySystem/src/main/java/tddmicroexercises/telemetrysystem/TelemetryClient.java

```diff
@@ -2,92 +2,41 @@
 
 import java.util.Random;
 
-public class TelemetryClient
+public class TelemetryClient implements TelemetryClientInterface
 {
     public static final String DIAGNOSTIC_MESSAGE = "AT#UD";
 
     private boolean onlineStatus;
-    private String diagnosticMessageResult = "";
 
-    private final Random connectionEventsSimulator = new Random(42);
+    private ConnectionInterface connection;
+
+    private TransferInterface transfer;
+
+    private String diagnosticMessageResult = "";
 
     public boolean getOnlineStatus()
     {
         return onlineStatus; 
     }
 
-    public void connect(String telemetryServerConnectionString)
-    {
-        if (telemetryServerConnectionString == null || "".equals(telemetryServerConnectionString))
-        {
-            throw new IllegalArgumentException();
-        }
-
-        // simulate the operation on a real modem
-        boolean success = connectionEventsSimulator.nextInt(10) <= 8;
 
-        onlineStatus = success;
+    TelemetryClient( Connection connection,Transfer transfer) {

```

Comment:
Here instead of Connection and Transfer class it should be ConnectionInterface and TransferInterface